src/dns/forward.h

```diff
@@ -23,6 +24,22 @@ class LookupDetails;
 
 void Init(void);
 
+/// A DNS domain name as described in RFC 1034 and RFC 1035.
+///
+/// The object creator is responsible for removing any encodings (e.g., URI
+/// percent-encoding) other than ASCII Compatible Encoding (ACE; RFC 5890) prior
+/// to creating a DomainName object. Domain names are stored as dot-separated
+/// ASCII substrings, with each substring representing a domain name label.
+/// DomainName strings are suitable for creating DNS queries and byte-by-byte
+/// case-insensitive comparison with configured dstdomain ACL parameters.
+///
+/// Even though an empty domain name is valid in DNS, DomainName objects are
+/// never empty.
+///
+/// The first label of a DomainName object may be a "*" wildcard (RFC 9525
+/// Section 6.3) if and only if the object creator explicitly allows wildcards.
+using DomainName = SBuf;
```

Comment:
For now, this alias allows us to document Dns::DomainName properties/invariants without changing a lot of code to cope with a new dedicated "real" type. It might stay that way, but I would not be surprised if we will eventually convert this into a class to reduce chances of wrong information (e.g., an IP address or even address:port authority) sneaking into this object and screwing up ACL checks and such.

Technically, this type alias declaration is not a true C++ forward declaration. I think we should make an exception for this forward.h entry, at least for now, because there is no better place to put it. However, if others insist, we can add something like src/dns/elements.h and put it there.